### ISSUE TITLE
`Subscribe Pending` shown for all federated subscribes

### ISSUE THREAD

**Issue by dessalines** [LemmyNet/lemmy#2685]
@Nutomic I've noticed this exists for all federated subscribes, even ones that happened in the past.

**Comment by hanubeki** [comment]
Cancelling "pending" subscription and re-subscribe worked for me, except for narwhal.city one (which is lotide)

**Comment by informapirata** [comment]
> L'annullamento dell'abbonamento "in sospeso" e il nuovo abbonamento hanno funzionato per me, ad eccezione di narwhal.city one (che è lotide)

With Mastodon and Pleroma the problem is solved by removing the follow and putting it back on again. But with Misskey it doesn't seem to work

**Comment by Nutomic** [comment]
These are actually marked as pending in the db now. Which is strange because [this](https://github.com/LemmyNet/lemmy/blob/main/migrations/2022-11-21-204256_user-following/up.sql) is the only recent db migration that touches the column, and it sets pending = false, not true.

**Comment by dessalines** [comment]
The only place I can see in the code where it sets it to true, is [here](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/activities/following/follow.rs#L66), which seems correct. 

Maybe there used to be an issue with accepting a follow, and the UI didn't reflect this for some reason?

Either way, I'm not sure if the correct solution is to do a DB migration to correct those, or ask people to re-follow all their federated communities.

**Comment by Nutomic** [comment]
Maybe an old Lemmy version used to write `subscribed: None` in case of success.

I guess its best to ask people to refollow if it bothers them, otherwise we risk causing even more problems.

**Comment by JackFromWisconsin** [comment]
> I guess its best to ask people to refollow if it bothers them, otherwise we risk causing even more problems.

Does "Subscribe Pending" still act as a normal subscription, then?

**Comment by Nutomic** [comment]
If it worked before the upgrade then it should still work fine. Lemmy is only displaying the status wrong.

**Comment by SiskoUrso** [comment]
Just started an instance today and having this same issue. Is this only graphical issue and you are indeed subscribed?

**Comment by Nutomic** [comment]
@SiskoUrso If its a new instance then that suggests that federation isnt working properly. Go through the troubleshooting steps from documentation.

https://join-lemmy.org/docs/en/administration/troubleshooting.html#federation

**Comment by SiskoUrso** [comment]
> @SiskoUrso If its a new instance then that suggests that federation isnt working properly. Go through the troubleshooting steps from documentation.
> 
> https://join-lemmy.org/docs/en/administration/troubleshooting.html#federation

Guess I'm not sure where its going wrong, testing with the commands in the Federation troubleshooting seemed to all return a valid response. 

I am using NGINX Proxy Manager , I have custom locations setup for /api, /pictrs, /feeds, /nodeinfo, /.well-known all pointing to the correct ip and port 8536. Looked over the nginx template provided in another post but not sure how to use that in NPM. 

Everything else seems to be working except this , as far as I can tell. 

Any guidance is appreciated.

**Comment by Nutomic** [comment]
Check the server logs for any errors while following a remote community. You can grep for the domain name of the other instance. 

If you can post in remote communities and these posts show up normally on the other instance, then the problem might be that other instances cant deliver their actions to your instance. In this case the other instance should log some errors so you need to contact the admin.

**Comment by Zetaphor** [comment]
I'm also experiencing this on a brand new instance. Nearly all of the communities I've subscribed to are showing as pending, the total number of comments isn't matching in my instance and theirs, and I was able to get valid JSON from all of the troubleshooting steps.

I'm able to subscribe to one community on an instance, but getting subscription pending for another community on the same instance.

I've checked my logs and am not seeing any errors. I'm also not running a reverse proxy, I just setup with ansible as described in the wiki

**Comment by verymilan** [comment]
idk i keep having this one the big ones: lemmy.ml and beehaw – on feddit it almost never happens

**Comment by Nutomic** [comment]
If it only happens with big instances then its most likely this configuration: https://lemmy.ml/post/1216911

**Comment by Daxiongmao87** [comment]
I'm having this very issue on my new instance, however it doesn't matter how big or small the community is.  Every single one that isn't local is "Subscribe Pending."  Comments are 0, and all votes are 1. 

Communities are being fetched, I can see the posts, but that's it.  I changed RUST_LOG to debugging, and i'm seeing no errors.  I'm also not seeing a recurring task to fetch hot ranks which I believe is supposed to be every 5 minutes.

**Comment by Nutomic** [comment]
Try following communities from a few different instances, it could be a problem on the other side. If it works you should see one log line for "send activity" and one for receive activity. You can also try if comments in remote communities go through.

**Comment by Daxiongmao87** [comment]
Hmm I seem to have subscribed successfully, I'm still getting just "1" on all ranks, and 0 comments.  I understand comment aggregation isn't instant.  It's an hour for federated, is that correct? And 5 minutes for hot rank updates? 

If that's the case, should I not see hot rank scheduled task pop up in the lemmy backend logs every 5 minutes?

**Comment by Nutomic** [comment]
Yes you will only receive new content, not that which was created prior to subscribing.

Closing this issue because it was a problem with the upgrade from 0.16 to 0.17, not relevant to new installations.

**Comment by Zetaphor** [comment]
> Closing this issue because it was a problem with the upgrade from 0.16 to 0.17, not relevant to new installations.

This has been happening on my new installation of 0.17.4

> If it only happens with big instances then its most likely this configuration: https://lemmy.ml/post/1216911 

I'm still stuck at pending on a lemmy.ml community. My server has 1 user and my worker count is 128.
I tried unsubscribing and resubscribing and the issue persists.

**Comment by Zetaphor** [comment]
FWIW, I was finally today able to use the method of unsubscribe-resubscribe to get these marked as fully joined.

The only server I'm still stuck pending on is lemmy.ml communities.

**Comment by mirisbowring** [comment]
I also have the problem on a clean install. I can Post / comment to subscribed communities, but no new Posts or comments are fetched. Also the logs do not show any helpful information

**Comment by maltfield** [comment]
Update: My instance upgrade to v0.17.4 and I'm still getting `Subscribe Pending` on some communities. Unsubscribing & re-subscribing still doesn't change this. And I still don't have full visibility into all the content on that remote community.

@dessalines can you please re-open this issue as it's clearly not yet resolved?

**Comment by Zetaphor** [comment]
@maltfield Can you call out which servers specifically?

**Comment by maltfield** [comment]
This problem exists between multiple instances trying to subscribe to remote communities lemmy.ml

One example is monero.town (v0.17.4) subscribing to !vegan@lemmy.ml (v0.17.4)

**Comment by Zetaphor** [comment]
This is why I asked, lemmy.ml is pretty broken right now and has been for a while.
ALL of my subscriptions are working, except for those on lemmy.ml. I've also only intermittently been receiving new content from them. I don't know if this is a problem with the codebase as much as it is with that particular server

**Comment by maltfield** [comment]
I'd still say it's a bug if the user hitting the `subscribe` button results in `Subscribe Pending` that *never* resolves itself. If the instance is under load and this happens for some minutes or hours, fine. But (I'm totally ignorant of the architecture here), the subscription should be queued & retried such that eventually the Subscription takes.


Lemmy's communication between instances should be robust and able to gracefully recover from network outages or other faults. Currently this isn't happening, so it's a bug with Lemmy.

**Comment by RocketDerp** [comment]
On 0.74 with working federation for many communities, isn't just lemmy.ml remote subscribe that is failing, lemmy.one today is also giving me the trouble.

For lemmy server operators, this PostgreSQL query will tell you how many pendings are stuck on your site:

```
SELECT person_id, p.name AS username, community_id, c.name as community, i.domain, community_follower.published
			FROM community_follower
			inner join person p on p.id = community_follower.person_id
			inner join community c on c.id = community_follower.community_id
			inner join instance i on c.instance_id = i.id
			WHERE pending='t'
			ORDER BY community_follower.published
```


I suggest we open a new issue on this? Anyone second that?

**Comment by chrisafraser** [comment]
Im trying to spin up a new instance and am seeing the same behaviour. In my logs I was able to find messages like this for all servers that I cannot get subscriptions working

`2023-06-19T20:11:18.426743Z  INFO Worker{worker.id=06aa9ebe-1cab-42fb-ac4b-54bbe7954ba2 worker.queue=default worker.operation.id=fe75d47d-f50d-43d6-921f-795aa50a1b68 worker.operation.name=process}:Job{execution_id=83235752-79dd-4e42-a6f5-d6e32c2e95a9 job.id=ed8bcdbd-4e78-464e-9ae0-871f3d79fd92 job.name=SendActivityTask}: activitypub_federation::core::activity_queue: Target server https://lemmy.ca/inbox rejected https://lemmy.my-domain-redacted.ca/activities/follow/c4b74591-767e-42a0-a160-5023e67c77aa, aborting`

**Comment by RocketDerp** [comment]
> Im trying to spin up a new instance and am seeing the same behaviour.

I would split out 'new instance maiden voyage' from 'established instance that is getting federated data from other peer servers'. 

On a new install, you may not have Docker setup correctly for outbound, see issue #3167 for an example of a new install having trouble. Beyond Docker proxy issues, it could be something with your ISP or the peer ISP not allowing connection or otherwise interfering. The Lemmy site midwest.social is not very busy, you might want to use that for another validation test.

**Comment by RocketDerp** [comment]
30 hours ago, someone posted a Python script to show how to create Lemmy accounts in mass: https://lemmy.ml/post/1344773

Issue #3200 has been opened to bring back captchas.

**Comment by arjan-s** [comment]
> Im trying to spin up a new instance and am seeing the same behaviour. In my logs I was able to find messages like this for all servers that I cannot get subscriptions working
> 
> `2023-06-19T20:11:18.426743Z INFO Worker{worker.id=06aa9ebe-1cab-42fb-ac4b-54bbe7954ba2 worker.queue=default worker.operation.id=fe75d47d-f50d-43d6-921f-795aa50a1b68 worker.operation.name=process}:Job{execution_id=83235752-79dd-4e42-a6f5-d6e32c2e95a9 job.id=ed8bcdbd-4e78-464e-9ae0-871f3d79fd92 job.name=SendActivityTask}: activitypub_federation::core::activity_queue: Target server https://lemmy.ca/inbox rejected https://lemmy.my-domain-redacted.ca/activities/follow/c4b74591-767e-42a0-a160-5023e67c77aa, aborting`

I have this exact error and couldn't find any solution either in the docs, other issues or elsewhere online. Did you find out anything?

**Comment by chrisafraser** [comment]
> > Im trying to spin up a new instance and am seeing the same behaviour. In my logs I was able to find messages like this for all servers that I cannot get subscriptions working
> > `2023-06-19T20:11:18.426743Z INFO Worker{worker.id=06aa9ebe-1cab-42fb-ac4b-54bbe7954ba2 worker.queue=default worker.operation.id=fe75d47d-f50d-43d6-921f-795aa50a1b68 worker.operation.name=process}:Job{execution_id=83235752-79dd-4e42-a6f5-d6e32c2e95a9 job.id=ed8bcdbd-4e78-464e-9ae0-871f3d79fd92 job.name=SendActivityTask}: activitypub_federation::core::activity_queue: Target server https://lemmy.ca/inbox rejected https://lemmy.my-domain-redacted.ca/activities/follow/c4b74591-767e-42a0-a160-5023e67c77aa, aborting`
> 
> I have this exact error and couldn't find any solution either in the docs, other issues or elsewhere online. Did you find out anything?

I did! When I was originally setting up I set my instance I set my external host name to a non publicly resolvable address (I figured external to my cluster would be fine so I picked Lemmy.service.dc1.consul). After I changed external to host name and added public dns entry I was running into this error.

 I took a closer look at the output from the federation json curl tests and found the old dns name was still present in some fields. After dropping my postgres db and starting fresh federation worked for me.

**Comment by UnKnoWn-Consortium** [comment]
@chrisafraser Thank you very much. Your solution works. Federation works after me dropping the existing database and recreating from sketch. I have encountered the same issue and have spent the last weekend trying to fix it.

**Comment by MattTheTekie** [comment]
> @chrisafraser Thank you very much. Your solution works. Federation works after me dropping the existing database and recreating from sketch. I have encountered the same issue and have spent the last weekend trying to fix it.

Thank you!
